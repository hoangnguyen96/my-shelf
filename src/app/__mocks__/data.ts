export const DATA_BOOKS = [
  {
    title: "Don't Make Me Think",
    author: "Steve Krug",
    category: "SelfHelpBook",
    imageUrl: "https://i.ibb.co/QbQgVtG/book1.png",
    description: "description will update later",
    status: false,
    publicationYear: 2000,
    rating: 4.5,
    createdDate: "3 Sep 2024 9:07 PM",
    edition: "first",
    id: "1",
  },
  {
    title: "Rich Dad Poor Dad",
    author: "Robert T. Kiyosaki",
    category: "SelfHelpBook",
    imageUrl: "https://i.ibb.co/9qHTtQF/book2.png",
    description:
      "Steve Krug is a usability consultant who has more than 30 years of experience as a user advocate for companies like Apple, Netscape, AOL, Lexus, and others. Based in part on the success of his first book, Don't Make Me Think, he has become a highly sought-after speaker on usability design.",
    status: false,
    publicationYear: 1997,
    rating: 4.5,
    createdDate: "3 Sep 2024 8:47 PM",
    edition: "second",
    id: "2",
  },
  {
    title: "The Design of Everyday Things",
    author: "Don Norman",
    category: "TextBook",
    imageUrl: "https://i.ibb.co/LnSBcCM/book5.png",
    description:
      "Steve Krug is a usability consultant who has more than 30 years of experience as a user advocate for companies like Apple, Netscape, AOL, Lexus, and others. Based in part on the success of his first book, Don't Make Me Think, he has become a highly sought-after speaker on usability design.",
    status: false,
    publicationYear: 1988,
    rating: 4.5,
    createdDate: "3 Sep 2024 8:51 PM",
    edition: "fourth",
    id: "3",
  },
];

export const DATA_USER = [
  {
    username: "admin",
    email: "admin@gmail.com",
    password: "$2a$10$MwSZ9vWSX7DVJwJ4SF5Mzuj2zMFjY52XM9L11gGG0j79gybowaLfq",
    isAdmin: true,
    phone: "",
    bio: "",
    avatar: "https://i.ibb.co/RHMqQGr/man-1.png",
    favorites: ["3", "2", "1", "13", "5"],
    shelfBooks: ["1", "6"],
    userId: "3733403",
    id: "1",
  },
  {
    username: "hung",
    email: "hung1@gmail.com",
    password: "$2a$10$Mt6ruwOzt3nzX1.HpSt.OOzut.teXmLE74WclmkP.Nn.gzVQrlqri",
    isAdmin: false,
    phone: "",
    bio: "",
    avatar: "https://i.ibb.co/88X1WfZ/avatar-default.png",
    favorites: ["1"],
    shelfBooks: ["2", "3", "1"],
    userId: "8857134",
    id: "2",
  },
];
