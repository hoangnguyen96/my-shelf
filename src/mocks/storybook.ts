export const mockRouter = {
  pathname: "/",
  route: "/",
  query: {},
  asPath: "/",
  basePath: "",
  push: () => Promise.resolve(true),
  replace: () => Promise.resolve(true),
  reload: () => {},
  back: () => {},
  prefetch: () => Promise.resolve(),
  beforePopState: () => {},
  isFallback: false,
  isLocaleDomain: false,
  isReady: true,
  isPreview: false,
  forward: () => {},
  events: {
    on: () => {},
    off: () => {},
    emit: () => {},
  },
};

export const mockSession = {
  user: {
    isAdmin: true,
    email: "admin@gmail.com",
    id: "3733403",
    name: "admin",
    image: "https://i.ibb.co/RHMqQGr/man-1.png",
  },
  expires: "2024-12-31T23:59:59.999Z",
};
