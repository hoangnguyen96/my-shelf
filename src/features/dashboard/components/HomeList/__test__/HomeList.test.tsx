import { fireEvent, render } from "@testing-library/react";
import { DATA_BOOKS, DATA_USER } from "@app/mocks/data";
import { HomeList } from "..";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

jest.mock("../../../actions", () => ({
  updateUserById: jest.fn(),
}));

describe("Home List", () => {
  const props = {
    user: DATA_USER[0],
    list: DATA_BOOKS,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should render correctly snapshot", () => {
    const { container } = render(<HomeList {...props} />);

    expect(container).toMatchSnapshot();
  });

  it("Should handle update favorite cart", () => {
    const { getAllByTestId } = render(<HomeList {...props} />);

    const buttons = getAllByTestId("update-favorite-cart");

    fireEvent.click(buttons[0]);
  });

  it("Should handle update favorite cart when data user no favorite", () => {
    const { getAllByTestId } = render(
      <HomeList {...props} user={DATA_USER[1]} />
    );

    const buttons = getAllByTestId("update-favorite-cart");

    fireEvent.click(buttons[2]);
  });
});
