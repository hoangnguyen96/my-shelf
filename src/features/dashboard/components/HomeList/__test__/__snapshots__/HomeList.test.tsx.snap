// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home List Should render correctly snapshot 1`] = `
<div>
  <div
    class="css-1jtg6vd"
  >
    <div
      class="css-583j2r"
    >
      <a
        href="/preview/1"
      >
        <img
          alt="Don't make me think"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="172"
          src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FQbQgVtG%2Fbook1.png&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FQbQgVtG%2Fbook1.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FQbQgVtG%2Fbook1.png&w=384&q=75 2x"
          style="color: transparent;"
          width="130"
        />
      </a>
      <p
        class="chakra-text css-0"
      >
        Don't Make Me Think
      </p>
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-0"
        >
          Steve Krug
          ,
        </p>
        <p
          class="chakra-text css-0"
        >
          2000
        </p>
      </div>
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-0"
        >
          4.5
          /
        </p>
        <p
          class="chakra-text css-c5z1pm"
        >
          5
        </p>
      </div>
      <button
        aria-label="Heart icon"
        class="chakra-button css-ujxaga"
        data-testid="update-favorite-cart"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-iz364c"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-583j2r"
    >
      <a
        href="/preview/2"
      >
        <img
          alt="Don't make me think"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="172"
          src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F9qHTtQF%2Fbook2.png&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F9qHTtQF%2Fbook2.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F9qHTtQF%2Fbook2.png&w=384&q=75 2x"
          style="color: transparent;"
          width="130"
        />
      </a>
      <p
        class="chakra-text css-0"
      >
        Rich Dad Poor Dad
      </p>
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-0"
        >
          Robert T. Kiyosaki
          ,
        </p>
        <p
          class="chakra-text css-0"
        >
          1997
        </p>
      </div>
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-0"
        >
          4.5
          /
        </p>
        <p
          class="chakra-text css-c5z1pm"
        >
          5
        </p>
      </div>
      <button
        aria-label="Heart icon"
        class="chakra-button css-ujxaga"
        data-testid="update-favorite-cart"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-iz364c"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
      </button>
    </div>
    <div
      class="css-583j2r"
    >
      <a
        href="/preview/3"
      >
        <img
          alt="Don't make me think"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="172"
          src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FLnSBcCM%2Fbook5.png&w=384&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FLnSBcCM%2Fbook5.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FLnSBcCM%2Fbook5.png&w=384&q=75 2x"
          style="color: transparent;"
          width="130"
        />
      </a>
      <p
        class="chakra-text css-0"
      >
        The Design of Everyday Things
      </p>
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-0"
        >
          Don Norman
          ,
        </p>
        <p
          class="chakra-text css-0"
        >
          1988
        </p>
      </div>
      <div
        class="css-k008qs"
      >
        <p
          class="chakra-text css-0"
        >
          4.5
          /
        </p>
        <p
          class="chakra-text css-c5z1pm"
        >
          5
        </p>
      </div>
      <button
        aria-label="Heart icon"
        class="chakra-button css-ujxaga"
        data-testid="update-favorite-cart"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-iz364c"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
